diff/diff.go

```diff
@@ -56,11 +59,47 @@ type Comparer interface {
 type ApplyConfig struct {
 	// ProcessorPayloads specifies the payload sent to various processors
 	ProcessorPayloads map[string]*types.Any
+	// Labels is used to specify any additional metadata
+	Labels map[string]string
```

Comment:
That's was my original thought as well, but after looking into how the `ProcessorPayloads` are used it seemed more of a hack, because we're not technically making a change in stream processor and the payload isn't passed anywhere? I might also be misunderstanding the intent
We haven't merged `hcsshim` changes that need this yet, but what we want to do is to introduce some snapshotter labels that would enable conditionally pass additional options to the tar2ext4 converter [here](https://github.com/containerd/containerd/blob/master/diff/lcow/lcow.go#L158).